lang/cs/Org.Apache.REEF.IMRU/OnREEF/IMRUTasks/UpdateTaskHost.cs

```diff
@@ -164,24 +164,28 @@ public byte[] Call(byte[] memento)
 
         /// <summary>
         /// Task close handler.
-        /// If the closed event is sent from driver, set _shouldCloseTask to 1 so that to inform the Call() to stop at the end of the current iteration.
+        /// Set _shouldCloseTask to 1 so that to inform the Call() to stop at the end of the current iteration.
         /// Then waiting for the signal from Call method. Either it is signaled or after _enforceCloseTimeoutMilliseconds,
-        /// checks if the task has been stopped. If not, throw IMRUTaskSystemException to enforce the task to stop.
+        /// If the closed event is sent from driver, checks if the task has been stopped. If not, throw IMRUTaskSystemException 
+        /// to enforce the task to stop.
         /// </summary>
         /// <param name="closeEvent"></param>
         public void OnNext(ICloseEvent closeEvent)
         {
-            var msg = Encoding.UTF8.GetString(closeEvent.Value.Value);
-            if (closeEvent.Value.IsPresent() && msg.Equals(TaskManager.CloseTaskByDriver))
+            Interlocked.Exchange(ref _shouldCloseTask, 1);
```

Comment:
Yes, they are the same. 
We can write a separate CloseHandler for both tasks to share, but as it uses instance variables in the tasks, put the handler inside the task class is convenient.
Writing a base class for those two? We usually write flatten task to make it easy to read. 
Do you have any suggestions?